core/src/main/scala/org/ensime/core/SemanticHighlighting.scala

```diff
@@ -50,36 +50,41 @@ class SemanticHighlighting(val global: RichPresentationCompiler) extends Compile
           add(TypeParamSymbol)
         } else if (sym.hasFlag(PARAM)) {
           add(ParamSymbol)
-        } else if (sym.hasFlag(ACCESSOR)) {
-          val under = sym.accessed
-          if (under.isVariable) {
-            add(VarFieldSymbol)
-          } else if (under.isValue) {
-            add(ValFieldSymbol)
-          } else {
-            false
+        } else {
+          if (sym.isDeprecated) {
```

Comment:
I nested that `else`, because only at that point I'm sure that I can add the symbol as deprecated and I don't have to ask about deprecation inside every remaining `if`. Otherwise, I will have to ask about deprecation of particular symbols and the code gets verbose. WDYT?